src/pudl_rmi/deprish.py

```diff
@@ -366,13 +368,17 @@ def fill_in_df(self, df_to_fill, common_allocated):
             filled_df[f"book_reserve{suffix}"] = filled_df[
                 f"book_reserve{suffix}"
             ].fillna(
-                filled_df[f"plant_balance{suffix}"]
+                (filled_df[f"plant_balance{suffix}"] * (1 - filled_df.net_salvage_rate))
                 - (
                     filled_df[f"depreciation_annual_epxns{suffix}"]
                     * filled_df.remaining_life_avg
                 )
             )
-
+            filled_df[f"net_plant_balance{suffix}"] = filled_df[
+                f"net_plant_balance{suffix}"
+            ].fillna(
+                filled_df[f"plant_balance{suffix}"] - filled_df[f"book_reserve{suffix}"]
+            )
```

Comment:
Is there a reason you made empty columns first and then filled them in here instead of with the assign statement above?

Reply:
Yes, even though it is a little opaque, which is probably an indication of a design problem.

1. net plant balance is now a part of the `DOLLAR_COLS`, which is used several times throughout the transform process which expects all of these cols to exist
2. this filling-in of missing values happens twice (once before we allocate the common records and once after). because of that we don't want to null the values during this fill-in step because it could already have info in there